v25.4.0 GA Release Notes (#20878)
cockroachdb/docs: src/current/_includes/releases/v25.4/v25.4.0.md

  <tr>
   <td>
    <p class="feature-summary">Vector Indexing Improvements</p>
    <p class="feature-description" markdown="1">
      Vector indexes in CockroachDB now support table backfills, making it easy to add vector indexes to existing tables without requiring manual data rewrites or service interruption. This update enhances the usability of vector search features by allowing seamless indexing of pre-existing data. Additionally, support for both cosine distance and inner product has been added, giving users more flexibility in selecting similarity metrics that best suit their machine learning or AI workloads. Together, these improvements make it easier to build and scale intelligent applications directly on CockroachDB.
    </p>
   </td>
   <td>25.4</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
  </tr>
  <tr>
   <td>
    <p class="feature-summary">Stable SQL API and Controlled Access to Unsupported crdb_internal Tables</p>
    <p class="feature-description" markdown="1">

>       CockroachDB is introducing a stable and well-defined SQL API for system introspection, ensuring that customers and third-party tools can safely and reliably query cluster metadata and performance data in production environments.

Reviewer: nit

```suggestion
      CockroachDB introduces a stable, well-defined SQL API for system introspection. This API enables customers and third-party tools to safely and reliably query cluster metadata and performance data in production environments.
```

    </p>
    <p class="feature-description">Each Transaction Diagnostic Bundle includes:</p>
    <ul class="feature-description">
      <li>A Jaeger and text trace of the full transaction execution path.</li>
      <li>Individual statement diagnostic bundles for every statement executed within the transaction.</li>
    </ul>
   </td>
   <td>25.4</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
  </tr>
  <tr>
   <td>
    <p class="feature-summary">Structured and Consistent Logging Channels</p>
    <p class="feature-description" markdown="1">
      CockroachDB has updated structured and consistent logging channels across the system to improve log discoverability, production observability, and support troubleshooting.
    </p>
    <p class="feature-description">New channels introduced:</p>
    <ul class="feature-description">

>       <li>CHANGEFEED – dedicated channel for Change Data Capture (CDC) logs, migrated from DEV and TELEMETRY.</li>

Reviewer: nit, we do not want to use the word "migrate" in this context:

```suggestion
      <li>CHANGEFEED – dedicated channel for Change Data Capture (CDC) logs, previously routed to DEV and TELEMETRY.</li>
```

      <li>KV_EXEC – channel for KV execution logs, separated from general distribution logs.</li>
    </ul>
    <p class="feature-description">Rerouted existing logs:</p>
    <ul class="feature-description">
      <li>sampled_query logs moved from TELEMETRY → SQL_EXEC for better categorization.</li>
    </ul>
   </td>
   <td>25.4</td>
   <td class="icon-center">{% include icon-yes.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
   <td class="icon-center">{% include icon-no.html %}</td>
  </tr>
 </tbody>
</table>

<h4 id="v25-4-0-cloud">CockroachDB Cloud</h4>

<table markdown="1">
 <thead>
  <tr>
   <th class="center-align" colspan="1" rowspan="2">Feature</th>
   <th class="center-align" colspan="5" rowspan="1">Availability</th>
  </tr>
  <tr>
    <th colspan="1" rowspan="1">Ver.</th>
    <th colspan="1" rowspan="1" style="white-space: nowrap;">Self-hosted</th>
    <th colspan="1" rowspan="1">Advanced</th>
    <th colspan="1" rowspan="1">Standard</th>
    <th colspan="1" rowspan="1">Basic</th>
  </tr>
 </thead>
 <tbody>
  <tr>
   <td>

>     <p class="feature-summary">Metrics Viewer Role for CockroachDB Cloud</p>

Author: @florence-crl Please review this 1 cloud feature highlight. It's observability-related and I know you're familiar with it. Thank you.

    <p class="feature-description" markdown="1">
      This release introduces the Metrics Viewer role on CockroachDB Cloud, providing dedicated permissions for metrics collection and insights. This role is designed to enhance overall security by enabling granular access control to cluster performance data.

      The Metrics Viewer role grants read-only access to cluster metrics and related insights, without conferring any administrative or data manipulation privileges.
    </p>
    <p class="feature-description">Key benefits:</p>
    <ul class="feature-description">
      <li>Improved Security: Enables the principle of least privilege by allowing users or services to monitor cluster performance without broader access to sensitive data or configuration.</li>
      <li>Targeted Access: Provides focused access to operational metrics, empowering teams to monitor health and performance efficiently.</li>
      <li>Enhanced Collaboration: Facilitates secure collaboration by allowing different teams (e.g., SREs, developers) to access relevant metrics without over-provisioned permissions.</li>
    </ul>

>     <p class="feature-description">When combined with the existing Cluster Developer role, a user can effectively gain "Power Developer" like functionalities, enabling them to develop applications and monitor their performance comprehensively while maintaining a clear separation of privileges for security.</p>

Reviewer: ```suggestion
    <p class="feature-description">When combined with the Cluster Developer role, users effectively gain 'Power Developer'-like capabilities, allowing them to build applications and monitor performance while maintaining clear privilege separation for security.</p>
```
